runbotics-orchestrator/src/main/java/com/runbotics/repository/ProcessCollectionRepository.java

```diff
@@ -13,6 +13,38 @@
 @Repository
 public interface ProcessCollectionRepository extends JpaRepository<ProcessCollection, UUID>, JpaSpecificationExecutor<ProcessCollection> {
 
+    @Query(value =
+        "SELECT COUNT(*) FROM ProcessCollection pc WHERE pc.id = ?1"
+    )
+    int countCollectionById(UUID id);
+
+    @Query(value =
+        "SELECT COUNT(*) " +
+            "FROM ProcessCollection pc " +
+            "LEFT JOIN pc.users u " +
+            "WHERE pc.id = ?1 AND " +
+            "(pc.isPublic = true OR pc.createdBy = ?2 OR u = ?2)"
+    )
+    int countAvailableCollectionById(UUID id, User user);
+
+    @Query(value =
+        "SELECT pc " +
+            "FROM ProcessCollection pc " +
+            "LEFT JOIN pc.users u " +
+            "WHERE pc.parentId = ?1 AND " +
+            "(pc.isPublic = true OR pc.createdBy = ?2 OR u = ?2)"
```

Comment:
what's the difference here? pc.createdBy = ?2 OR u = ?2

Reply:
"createdBy" is related to a user, who is the creator of this collection. "u" is from relation between collections and users, so if a user will be in relation with collection in table process_collection_user then it means that the collection is shared for it.